addons/l10n_tw_edi_ecpay/tests/test_edi.py

```diff
@@ -237,45 +236,49 @@ def test_07_fail_data_validation(self):
         with self.assertRaises(UserError):
             send_and_print.action_send_and_print()
 
-    def test_08_invoice_with_downpayment(self):
-        """Ensure downpayment with -ve quantity is normalized for ECPay JSON."""
-        invoice = self.init_invoice(
-            'out_invoice', partner=self.partner_a, products=self.product_a,
-        )
-        invoice.write({
-            "invoice_line_ids": [
-                Command.create({
-                    "name": "Downpayment",
-                    "price_unit": 10.0,
-                    "quantity": -1.0,
-                    "tax_ids": self.tax_sale_a,
-                }),
-            ],
-        })
-        invoice.action_post()
-
-        self.assertListEqual(
-            invoice._l10n_tw_edi_generate_invoice_json()['Items'],
-            [
-                {
-                    'ItemSeq': 1,
-                    'ItemName': 'product_a',
-                    'ItemCount': 1.0,
-                    'ItemWord': 'Units',
-                    'ItemPrice': 1000.0,
-                    'ItemTaxType': '1',
-                    'ItemAmount': 1000.0
-                },
-                {
-                    'ItemSeq': 2,
-                    'ItemName': 'Downpayment',
-                    'ItemCount': 1.0,
-                    'ItemWord': False,
-                    'ItemPrice': -10.0,
-                    'ItemTaxType': '1',
-                    'ItemAmount': -10.0
-                },
-            ],
+    def test_08_refund_invoice_with_payment(self):
+        """
+        This tests the flow of refunding an invoice that has the payment has been registered and
+        has already been sent to the ECpay platform with an offline agreement type.
+        Make sure that the credit note is created with the correct fields and values from the wizard
+        and the credit note is successfully sent to the ECpay platform
+        """
+        send_and_print = self.create_send_and_print(self.basic_invoice)
+        with patch(CALL_API_METHOD, new=self._test_08_mock):
+            basic_invoice_json = self.basic_invoice._l10n_tw_edi_generate_invoice_json()
+            send_and_print.action_send_and_print()
+            # register payment
+            self.env['account.payment.register'].with_context(active_model='account.move', active_ids=self.basic_invoice.ids).create({
+                'amount': self.basic_invoice.amount_total,
+            })._create_payments()
+            self.assertEqual(self.basic_invoice.payment_state, 'in_payment')
```

Comment:
This is only true if you install the accountant app.

```suggestion
            self.assertEqual(self.basic_invoice.payment_state, self.env['account.move']._get_invoice_in_payment_state())
```